OsmAnd-java/src/main/java/net/osmand/render/RenderingRuleSearchRequest.java

```diff
@@ -348,4 +377,70 @@ public String toString() {
 		}
 		return builder.toString();
 	}
-}
\ No newline at end of file
+
+	@Nullable
+	public String searchIconByTags(@NotNull Map<String, String> tags) {
+		return searchBestPropertyByTransformedTags(tags, RenderingRulesStorage.POINT_RULES,
+				storage.PROPS.R_ICON, storage.PROPS.R_ICON_ORDER);
+	}
+
+	@Nullable
+	private String searchBestPropertyByTransformedTags(@NotNull Map<String, String> transformedTags, int rulesNumber,
```

Comment:
why do we need transformed tags? is it wrong name?

Reply:
what does it mean best? all comparatives in the code best / better is a bad approach. method should directly state what's sorting about, first rule or max tags matching

Reply:
searchTopOrderedPropertyByTags